### ISSUE TITLE
Salt Onedir missing standard python module 'cmath'

### ISSUE THREAD

**Issue by carterkoyle** [saltstack/salt#63718]
**Description**
Pythons standard '[cmath](https://github.com/python/cpython/blob/3.11/Doc/library/cmath.rst)' library is not included in Salts onedir python in v3005.1

**Setup**
- [x] Rocky9 VM running in GCP
- [x] onedir packaging
- [x] used bootstrap to install

**Steps to Reproduce the behavior**
Deploy a Rocky9 virtual machine in GCP and install onedir salt using the commands below:
```
sudo rpm --import https://repo.saltproject.io/salt/py3/redhat/9/x86_64/3005/SALTSTACK-GPG-KEY2.pub
curl -fsSL https://repo.saltproject.io/salt/py3/redhat/9/x86_64/3005.repo | sudo tee /etc/yum.repos.d/salt.repo
sudo yum install salt-master
sudo yum install salt-minion
sudo systemctl enable salt-master && sudo systemctl start salt-master
sudo systemctl enable salt-minion && sudo systemctl start salt-minion
```

Use 'salt-pip' to install the pandas package
```
salt-pip install pandas
```
Create a new runner `/srv/salt/_runners/sample_runner.py` that imports the 'pandas' library. Pandas requires the cmath. The contents of this new sample runner should look something like this:

```
import pandas

def sample_runner():
    print("this is just a sample. The runner will fail to import pandas due to onedir python missing 'cmath'")
```

Run a sync_all to ensure the runner is recognized
```
salt-run saltutil.sync_all
```

Execute the runner.
```
salt-run sample_runner.sample_runner
```

**Expected behavior**
The sample runner successfully imports pandas. The cmath module is included in onedirs 'path', so that it is available to any/all runners 

**Actual behavior**
```
[root@sm1 carterkoyle]# salt-run sample_runner.sample_runner
'sample_runner' __virtual__ returned False: No module named 'cmath'
```

**Versions Report**
```
[root@sm1 carterkoyle]# salt --versions-report
Salt Version:
          Salt: 3005.1

Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.0
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.21
      pycrypto: 3.17
  pycryptodome: 3.9.8
        pygit2: Not Installed
        Python: 3.9.16 (main, Nov  1 2022, 00:00:00)
  python-gnupg: 0.5.0
        PyYAML: 5.4.1
         PyZMQ: 23.2.0
         smmap: Not Installed
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4

System Versions:
          dist: rocky 9.0 Blue Onyx
        locale: utf-8
       machine: x86_64
       release: 5.14.0-70.26.1.el9_0.x86_64
        system: Linux
       version: Rocky Linux 9.0 Blue Onyx
```

**Additional context**
We are currently working around this problem by using the state pasted below. This did require us to place a static version of the cmath module in our own salt repo, but it works fine as a temporary workaround

```
# Salts python onedir does NOT ship with the cmath module, but some of our custom runners use pandas, which requires cmath, so we need to make sure that it is available.
{%- if grains.pythonexecutable == '/opt/saltstack/salt/run/run' %}
rhel_ensure_cmath_in_onedir_python_path:
  file.managed:
    - name: /opt/saltstack/salt/run/lib-dynload/cmath.cpython-39-x86_64-linux-gnu.so
    - source: salt://_files/comp/rhel_base/cmath.cpython-39-x86_64-linux-gnu.so
    - user: root
    - group: root
    - mode: 0755
{%- endif %}
```

**Comment by s0undt3ch** [comment]
Hello @carterkoyle.

The onedir tools/approach is going to change from `3005.x` to `3006.x`.
We just released `3006.0rc1`.
Can I ask you to confirm if you still have the same problem with 3006.0rc1?
I'm 99% certain that you wont, but I'd prefer confirmation, of course.

**Comment by s0undt3ch** [comment]
Confirmed.
The upcoming release no longer suffers from this issue.
```
❯ wget https://staging.repo.saltproject.io/salt_rc/salt/py3/onedir/3006/salt-3006.0rc1-onedir-linux-x86_64.tar.xz
--2023-03-09 10:57:47--  https://staging.repo.saltproject.io/salt_rc/salt/py3/onedir/3006/salt-3006.0rc1-onedir-linux-x86_64.tar.xz
Loaded CA certificate '/etc/ssl/certs/ca-certificates.crt'
Resolving staging.repo.saltproject.io (staging.repo.saltproject.io)... 13.225.244.123, 13.225.244.118, 13.225.244.20, ...
Connecting to staging.repo.saltproject.io (staging.repo.saltproject.io)|13.225.244.123|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 60368488 (58M) [binary/octet-stream]
Saving to: ‘salt-3006.0rc1-onedir-linux-x86_64.tar.xz’

salt-3006.0rc1-onedir-linux-x86_64.tar.xz             100%[======================================================================================================================>]  57,57M  21,3MB/s    in 2,7s

2023-03-09 10:57:51 (21,3 MB/s) - ‘salt-3006.0rc1-onedir-linux-x86_64.tar.xz’ saved [60368488/60368488]

/tmp/test
❯ tar xf salt-3006.0rc1-onedir-linux-x86_64.tar.xz
/tmp/test
❯ salt/bin/python3
Python 3.10.10 (main, Feb 23 2023, 05:21:34) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import cmath
>>>
```
